R/derive_param_computed.R

```diff
@@ -147,48 +196,246 @@
+#' expressions. If it is a character vector, it converts it to a list of
+#' symbols.
+#'
+#' @param parameters The argument to check
+#'
+#' @param optional Is the checked argument optional? If set to `FALSE` and
+#'   `parameters` is `NULL` then an error is thrown.
+#'
+#' @return The `parameters` argument (converted to a list of symbol, if it is a
+#'   character vector)
+#'
+#' @keywords other_advanced
+#' @family other_advanced
+assert_parameters_argument <- function(parameters, optional = TRUE) {
+  assert_logical_scalar(optional)
+  if (optional && is.null(parameters)) {
+    return(invisible(parameters))
+  }
+
+  if (typeof(parameters) == "character") {
+    parameters <- map(parameters, sym)
+  } else {
+    if (!inherits(parameters, "list") || any(!map_lgl(
+      parameters,
+      ~ is_call(.x) || is_expression(.x)
+    ))) {
+      abort(
+        paste0(
+          "`",
+          arg_name(substitute(parameters)),
+          "` must be a character vector or a list of expressions but it is ",
+          what_is_it(parameters),
+          "."
+        )
+      )
+    }
+  }
+  parameters
+}
+
+#' Creating Temporary Parameters and `<variable>.<parameter>` Variables
+#'
+#' The function creates temporary parameters and variables of the form
+#' `<variable>.<parameter>`, e.g., `AVAL.WEIGHT`.
+#'
+#' @param dataset Input dataset
+#'
+#' @param by_vars By variables
+#'
+#' @param parameters List of parameter codes
+#'
+#'   The input dataset is restricted to the specified parameter codes. If an
+#'   expression is specified, a new parameter code is added to the input
+#'   dataset. The name of the element defines the parameter code and the
+#'   expression the observations to select.
+#'
+#'   *Permitted Values:* A character vector of `PARAMCD` values or a list of expressions
+#'
+#' @param analysis_value
+#'
+#'   All variables of the form `<variable>.<parameter>` like `AVAL.WEIGHT` are
+#'   added to the input dataset. They are set to the value of the variable for
+#'   the parameter. E.g., `AVAL.WEIGHT` is set to the value of `AVAL` where
+#'   `PARAMCD == "WEIGHT"`.
+#'
+#'   *Permitted Values:* An unquoted expression
+#'
+#' @param filter Filter condition used for restricting the input dataset
+#'
+#'    The specified filter condition is used in the warnings only. It is not
+#'    applied to the input dataset.
+#'
+#' @return A dataset with one observation per by group. It contains the
+#'   variables specified for `by_vars` and all variables of the form
+#'   `<variable>.<parameter>` occurring in `analysis_value`.
+#'
+#' @keywords other_advanced
+#' @family other_advanced
+get_hori_data <- function(dataset,
+                          by_vars,
+                          parameters,
+                          analysis_value,
+                          filter) {
+  assert_vars(by_vars)
+  assert_data_frame(dataset, required_vars = by_vars)
+  assert_parameters_argument(parameters)
+  assert_expr(analysis_value)
```

Comment:
```suggestion
```
should we change the input check for `analysis_value`? -
>An unquoted expression

Reply:
Updated

Reply:
Just curious how do we check this type of input - "unquoted expression"?

Reply:
I think we can not check it because if something invalid like `AVAL +` is specified, an error is issued before the function is executed.